airflow/models/dagbag.py

```diff
@@ -166,6 +166,36 @@ def dag_ids(self) -> List[str]:
         """
         return list(self.dags.keys())
 
+    @provide_session
+    def has_dag(self, dag_id: str, session: Session = None):
+        """
+        Checks the "local" cache, if it exists, and is not older than the confiured cache time, return True
+        else check in the DB if the dag exists, return True, False otherwise
+        """
+        if dag_id in self.dags:
+            min_serialized_dag_fetch_secs = timedelta(seconds=settings.MIN_SERIALIZED_DAG_FETCH_INTERVAL)
+            if dag_id in self.dags_last_fetched:
+
+                return timezone.utcnow() < self.dags_last_fetched[dag_id] + min_serialized_dag_fetch_secs
```

Comment:
This is the wrong behaviour -- if the dag still exists, but it just hasn't been fetched in this DagBag instance it would return false.

Reply:
How do you suggest I resolve this? My thinking was that once the dag exists in cache, it'll still exist in dags_last_fetched?

Reply:
paging @kaxil

Reply:
Looking at how that is used, nothing in the DagBag deals with the case where a dagmodel/serialized dag row is deleted -- and the cached version would be returned, or maybe an error would be thrown.

EIther way, it doesn't really handle this case form what I can tell.

Reply:
Once https://github.com/apache/airflow/pull/16368 is merged, you can just call `get_dag(dag_id)`, if it returns `None`, you can return False